App/Sagas/TextileSagas.js

```diff
@@ -239,24 +239,52 @@ export function * shareImage ({thread, hash, caption}) {
 
 export function * photosTask () {
   try {
-    const photos = yield call(queryPhotos)
-    for (const photo of photos.reverse()) {
+    const camera = yield select(TextileSelectors.camera)
+    const processed = camera && camera.processed ? camera.processed : []
+
+    let allPhotos = yield call(getAllPhotos)
+    if (camera === undefined || camera.processed === undefined || processed.length === 0) {
+      const ignoredPhotos = allPhotos.splice(1)
+      for (const uri of ignoredPhotos.map((photo) => photo.uri)) {
```

Comment:
No need to loop through all the photos twice, just loop through `ingnoredPhotos` and `yield put(TextileActions.imageIgnore(photo.uri))` (the map is the additional loop through)

Reply:
cool